### PR TITLE
Eg-321, retrieve scratch process from external-guidance

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/config/AppConfig.scala [hunk 1]
```diff
@@ -32,9 +43,9 @@ class AppConfig @Inject() (val config: Configuration, servicesConfig: ServicesCo
   val assetsPrefix: String = assetsUrl + config.get[String]("assets.version")
   val analyticsToken: String = config.get[String](s"google-analytics.token")
   val analyticsHost: String = config.get[String](s"google-analytics.host")
-
   val reportAProblemPartialUrl: String = s"$contactBaseUrl/contact/problem_reports_ajax?service=$serviceIdentifier"
   val reportAProblemNonJSUrl: String = s"$contactBaseUrl/contact/problem_reports_nonjs?service=$serviceIdentifier"
-
   val languageMap: Map[String, Lang] = ListMap("english" -> Lang("en"), "cymraeg" -> Lang("cy"))
+
+  lazy val externalGuidanceBaseUrl: String = servicesConfig.baseUrl("external-guidance")
```
Comment (woollett-r) [inline app/config/AppConfig.scala]: IntelliJ is reporting some trailing whitespace on this line.

File: app/connectors/httpParsers/GetScratchProcessHttpParser.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        case None =>
+          logger.error("Unable to parse successful response when requesting process")
+          Left(InternalServerError)
+      }
+    case (_, _, response) if response.status == NOT_FOUND =>
+      Left(InvalidProcessError)
+    case (_, _, response) if response.status == BAD_REQUEST =>
+      Left(InvalidProcessError)
+    case unknown =>
+      logger.info(s"unexpected ${unknown} response received when requesting process")
```
Comment (woollett-r) [inline app/connectors/httpParsers/GetScratchProcessHttpParser.scala]: IntelliJ is suggesting the enclosing curly braces are not required.

File: test/connectors/GuidanceConnectorSpec.scala [hunk 3]
```diff
@@ -18,31 +18,42 @@ package connectors
...
+import mocks.{MockAppConfig, MockHttpClient}
 import scala.concurrent.ExecutionContext.Implicits.global
-
-class GuidanceConnectorSpec extends BaseSpec {
+import play.api.test.{DefaultAwaitTimeout, FutureAwaits}
+import play.api.libs.json.Json
+import models.ocelot.Process
+import scala.concurrent.Future
+
+class GuidanceConnectorSpec extends BaseSpec with MockHttpClient {
```
Comment (woollett-r) [inline test/connectors/GuidanceConnectorSpec.scala]: Both class and test trait extend MockHttpClient. Can probably just define on trait.

File: test/connectors/GuidanceConnectorSpec.scala [hunk 4]
```diff
@@ -18,31 +18,42 @@ package connectors
...
   }
 
   "Calling the scratchProcess with an existing scratch process UUID" should {
 
-    "return a model representing the Ocelot Scatch Process" in {
-      val hc: HeaderCarrier = HeaderCarrier()
-      val target = new GuidanceConnector()
-      val expectProcess = Some(target.scratchStubbedProcess)
-      val result = target.scratchProcess("683d9aa0-2a0e-4e28-9ac8-65ce453d2730")(hc, implicitly)
+    "return a model representing the Ocelot Scatch Process" in new Test {
```
Comment (woollett-r) [inline test/connectors/GuidanceConnectorSpec.scala]: The connector handles both successful and unsuccessful calls so a test for the later would be good.

File: test/controllers/HelloWorldControllerSpec.scala [hunk 5]
```diff
@@ -35,7 +35,7 @@ class HelloWorldControllerSpec extends WordSpec with Matchers with GuiceOneAppPe
     private val configuration = Configuration.load(env)
 
     private val serviceConfig = new ServicesConfig(configuration, new RunMode(configuration, Mode.Dev))
-    private val appConfig = new AppConfig(configuration, serviceConfig)
+    private val appConfig = new AppConfigImpl(configuration, serviceConfig)
```
Comment (woollett-r) [inline test/controllers/HelloWorldControllerSpec.scala]: Following the definition of a mock AppConfig these tests can be simplified avoiding the necessity to create the environment components required to create a real AppConfig instance. See the equivalent test in manage-external-guidance-frontend.

File: test/controllers/SwitchLanguageControllerSpec.scala [hunk 6]
```diff
@@ -27,15 +27,15 @@ import play.api.mvc.Cookies
 import play.api.i18n.{MessagesApi,DefaultLangsProvider}
 import uk.gov.hmrc.play.bootstrap.config.{RunMode, ServicesConfig}
 import uk.gov.hmrc.play.bootstrap.tools.Stubs._
-import config.AppConfig
+import config.AppConfigImpl
```
Comment (woollett-r) [inline test/controllers/SwitchLanguageControllerSpec.scala]: We should be able to use the mock configuration here also.

File: test/controllers/HelloWorldControllerSpec.scala (hunk 5 above)
Comment (cmrhmrc) [reply test/controllers/HelloWorldControllerSpec.scala]: This the HelloWorld test and will most likely dissappear at some point

**Pull request merged**